### DISCUSSION TITLE
How to qualify inside the dataset_module

### DISCUSSION THREAD

**Discussion by davekaro** [jeremyevans/sequel#2038] [General]
I have a dataset_module defined on two different Sequel::Model classes (User and School) like this:

```
dataset_module do
  where :kept, discarded_at: nil
end
```

Now, if I join like `User.join(:schools, id: :school_id).kept.first` I get an expected error "column reference "discarded_at" is ambiguous (PG::AmbiguousColumn)". I know I could use `qualify` at the call site, like `User.join(:schools, id: :school_id).kept.qualify.first`, but can I qualify only the `discarded_at` and still use the `where` method of the dataset_module? Best I could come up with was to assign the table_name to a local variable like this:

```
model_table_name = table_name

dataset_module do
  where :kept, Sequel[model_table_name][:discarded_at] => nil
end
```

Any other ideas?

**Comment by davekaro** [comment]
One thing I noticed is the model's `dataset` methods includes `model` (http://sequel.jeremyevans.net/rdoc/classes/Sequel/Model/DatasetMethods.html#method-i-model), but the dataset_module (http://sequel.jeremyevans.net/rdoc/classes/Sequel/Model/DatasetModule.html) does not, though it does store an instance variable referencing the model (`@model`). Maybe if that was exposed as an attr_reader or similar, then I could do:

```
dataset_module do
  where :kept, Sequel[model.table_name][:discarded_at] => nil
end
```

**Reply by jeremyevans** [reply]
Seems like it should be fine to add `attr_reader :model` in `Sequel::Model::DatasetModule`.  If you want to work on a patch for that (with specs), that would be great.  Otherwise, I can probably work on it before the next release.
